src/WpfMath/UserDefinedColorParser.cs

```diff
@@ -0,0 +1,264 @@
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        private static ColorStringTypes GetColorString(string input)
+        {
+            if (IsByteHexTrain(input,6)==true)
+            {
+                return ColorStringTypes.Hex_shortString;
+            }
+            else if (IsByteHexTrain(input,8))
+            {
+                return ColorStringTypes.Hex_longString;
+            }
+            else if (IsByteTrain(input, 3) == true)
+            {
+                return ColorStringTypes.Byte_shortString;
+            }
+            else if (IsByteTrain(input, 4) == true)
+            {
+                return ColorStringTypes.Byte_longString;
+            }
+            else
+            {
+                return ColorStringTypes.None;
+            }
+        }
+
+        /// <summary>
+        /// Returns a value that tells if the <paramref name="input"/> contains byte values, separated by a ",".
+        /// </summary>
+        /// <example>
+        /// 23,78,56=>true
+        /// 789,fp=>false
+        /// </example>
+        /// <param name="input"></param>
+        /// <param name="num"></param>
+        /// <returns></returns>
+        private static bool IsByteTrain(string input, int num)
+        {
+            bool StrCheck = false;
+            string[] arrStr = input.Trim().Split(',');
+            if (num == arrStr.Length)
+            {
+                int i = 0;
+                foreach (var item in arrStr)
+                {
+                    if (Byte.TryParse(item, out byte result) == true)
+                    {
+                        i += 1;
+                    }
+                    else { continue; }
+                }
+                StrCheck = i == num ? true : false;
+            }
+            else
+            {
+                StrCheck = false;
+            }
+            return StrCheck;
+        }
+        
+        /// <summary>
+        /// Returns a value that tells if the <paramref name="input"/> contains byte hex values,
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="num"></param>
+        /// <returns></returns>
+        /// <remarks><paramref name="input"/> should be left in the raw state(e.g.;#56e245, not 56e245).</remarks>
+        private static bool IsByteHexTrain(string input, int num)
+        {
+            bool StrCheck = false;
+            string subStr = input.Substring(1);
+            if (num == subStr.Length)
+            {
+                int c = 0;
+                for (int i = 1; i < num; i+=2)
+                {
+                    string item = subStr[i - 1].ToString() + subStr[1].ToString();
+                    if (Byte.TryParse(item, System.Globalization.NumberStyles.AllowHexSpecifier, CultureInfo.InvariantCulture, out byte result) == true)
+                    {
+                        c+=2;
```

Comment:
It seems to me like this approach will consider cases like `33XX3333` as valid. Please fix it.

Reply:
I tried #33XX3333  but it doesn't work

Reply:
Okay, you're right. I haven't noticed the check `if (num == subStr.Length)`.